clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -303,6 +302,19 @@ private void throwIfOffsetOutOfRange() throws OffsetOutOfRangeException {
     }
 
     /**
+     * If any topic from previous fetchResponse contatains Authorization error, throw ApiException.
+     * @throws ApiException
+     */
+    private void throwIfUnauthorized() throws ApiException {
```

Comment:
To clarify, you mean topic metadata, right? I don't think consumer metadata is an issue here.